x-pack/plugin/esql/qa/testFixtures/src/main/resources/date.csv-spec

```diff
@@ -614,3 +614,31 @@ birth_date:datetime
 1952-02-27T00:00:00.000Z
 1953-04-21T00:00:00.000Z
 ;
+
+dateMathAddDatePeriodsThenAddToDatetime
+row n = to_dt("1953-04-04T00:00:00.000Z") | eval then = (4 years + 3 months + 2 weeks + 1 day) + n | keep then;
+
```

Comment:
Please add some tests without parenthesis just to be sure.

Reply:
I guess the parentheses are needed to trigger the type-specific folding. There are tests without parentheses already.

Reply:
The parentheses are actually redundant. I just wanted tests that make it obvious that the expression adds the durations/periods _before_ adding to a datetime, which is the currently unsupported case.

Reply:
Added more test cases! Also added tests for subtractions from datetimes combined with first folding period/duration constants.